tests/playwright-test/test-server.spec.ts

```diff
@@ -328,3 +328,37 @@ test('pauseOnError no errors', async ({ startTestServer, writeFiles }) => {
   expect(await testServerConnection.runTests({ pauseOnError: true, locations: [] })).toEqual({ status: 'passed' });
   expect(testServerConnection.events.filter(e => e[0] === 'testPaused')).toEqual([]);
 });
+
+test('runGlobalSetup returns env', async ({ startTestServer, writeFiles }) => {
+  await writeFiles({
+    'playwright.config.ts': `
+      export default { globalSetup: './global-setup.ts' };
+    `,
+    'global-setup.ts': `
+      export default async function() {
+        delete process.env.MAGIC_BEFORE;
+        process.env.MAGIC_AFTER = '43';
```

Comment:
try incrementing MAGIC_AFTER instead of setting to see how the two runs influence each other